packages/core/src/lib/context/fragments/dom.ts

```diff
@@ -0,0 +1,63 @@
+import { getContext, onMount, setContext } from 'svelte'
+import { currentWritable, toCurrentReadable, type CurrentReadable } from '../../utilities'
+
+export type Size = {
+  width: number
+  height: number
+}
+
+type DOMContext = {
+  /** The canvas wrapper element */
+  dom: HTMLElement
+  size: CurrentReadable<Size>
+}
+
+export type CreateDOMContextOptions = {
+  dom: HTMLElement
+}
+
+export const createDOMContext = (options: CreateDOMContextOptions) => {
+  const domRect = options.dom.getBoundingClientRect()
+
+  let lastWidth = domRect.width
+  let lastHeight = domRect.height
+
+  const size = currentWritable({ width: domRect.width, height: domRect.height })
```

Comment:
I think it would be nice if we could forward other `Rect` properties to size as well. Cause at points getting more than width/height is helpful.